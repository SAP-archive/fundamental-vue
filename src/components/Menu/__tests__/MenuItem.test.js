import { mount, createLocalVue } from "@vue/test-utils";
import FdMenuItem from "./../MenuItem.vue";
import FdMenuLink from "./../MenuLink.vue";
import FundamentalVue from "./../../../index";

const mountMenuItem = template => {
  const localVue = createLocalVue();
  localVue.use(FundamentalVue);

  return mount(
    {
      template,
      components: { FdMenuItem, FdMenuLink }
    },
    {
      localVue,
      provide: {
        menuList: {
          canHaveAddon: false,
          menuItemDidClick() {}
        },
        menuHighlight: { highlightedId: null },
        menu: {
          registerMenuItem() {},
          unregisterMenuItem() {}
        }
      }
    }
  );
};

describe("MenuItem", () => {
  it("emits click-event", async () => {
    const wrapper = mountMenuItem(`<FdMenuItem>hello</FdMenuItem>`);
    wrapper.find("a").trigger("click");
    await wrapper.vm.$nextTick();
    const clicks = wrapper.find(FdMenuItem).emitted("click");
    expect(clicks).toHaveLength(1);
  });

  it("does render embeddded link with href", () => {
    const wrapper = mountMenuItem(`
    <FdMenuItem uid="aaa">
      <FdMenuLink href='#'>Item 1</FdMenuLink>
    </FdMenuItem>`);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("does render embedded link", () => {
    const wrapper = mountMenuItem(
      `
      <FdMenuItem uid="aaa">
        <FdMenuLink>Item 1</FdMenuLink>
      </FdMenuItem>`
    );
    expect(wrapper.element).toMatchSnapshot();
  });

  it("does render selected state", () => {
    const wrapper = mountMenuItem(
      `
      <FdMenuItem uid="aaa" :selected="true">
        <FdMenuLink>Item 1</FdMenuLink>
      </FdMenuItem>`
    );
    expect(wrapper.element).toMatchSnapshot();
  });
});
