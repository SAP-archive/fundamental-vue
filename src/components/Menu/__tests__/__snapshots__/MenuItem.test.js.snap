// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem does render embeddded link with href 1`] = `
<li>
  <!---->
   
  <a
    class="fd-menu__item"
    href="#"
  >
    Item 1
  </a>
</li>
`;

exports[`MenuItem does render embedded link 1`] = `
<li>
  <!---->
   
  <a
    class="fd-menu__item"
  >
    Item 1
  </a>
</li>
`;

exports[`MenuItem does render selected state 1`] = `
<li>
  <!---->
   
  <a
    class="fd-menu__item"
  >
    Item 1
  </a>
</li>
`;
